aggregator/src/datastore.rs

```diff
@@ -1241,19 +1244,63 @@ impl<C: Clock> Transaction<'_, C> {
             .execute(
                 &stmt,
                 &[
-                    /* task_id */ &report.task_id().get_encoded(),
-                    /* report_id */ &report.metadata().id().get_encoded(),
-                    /* client_timestamp */ &report.metadata().time().as_naive_date_time()?,
+                    /* task_id */ &new_report.task_id().get_encoded(),
+                    /* report_id */ &new_report.metadata().id().get_encoded(),
+                    /* client_timestamp */
+                    &new_report.metadata().time().as_naive_date_time()?,
                     /* extensions */ &encoded_extensions,
                     /* public_share */ &encoded_public_share,
                     /* leader_input_share */ &encoded_leader_share,
                     /* helper_encrypted_input_share */ &encoded_helper_share,
                 ],
             )
             .await?;
-        if rows_affected == 0 {
+
+        if rows_affected > 1 {
+            return Err(Error::DbState(format!(
```

Comment:
I guess this should be a panic, since it's impossible/indicates a bug in Postgres. (or just drop this check entirely)